tests/integration/suite/daprd/serviceinvocation/http/basic.go

```diff
@@ -114,38 +122,43 @@ func (b *basic) Run(t *testing.T, ctx context.Context) {
 			return resp.StatusCode, string(body)
 		}
 
-		for _, ts := range []struct {
+		for i, ts := range []struct {
 			url     string
 			headers map[string]string
 		}{
 			{url: fmt.Sprintf("http://localhost:%d/v1.0/invoke/%s/method/foo", b.daprd1.HTTPPort(), b.daprd2.AppID())},
 			{url: fmt.Sprintf("http://localhost:%d/v1.0/invoke/%s/method/foo", b.daprd2.HTTPPort(), b.daprd1.AppID())},
 			{url: fmt.Sprintf("http://localhost:%d/v1.0////invoke/%s/method/foo", b.daprd2.HTTPPort(), b.daprd1.AppID())},
 			{url: fmt.Sprintf("http://localhost:%d/v1.0//invoke//%s/method//foo", b.daprd1.HTTPPort(), b.daprd2.AppID())},
-			{url: fmt.Sprintf("http://localhost:%d///foo", b.daprd1.HTTPPort()), headers: map[string]string{
+			// We cannot use `///foo` here because the test app uses the standard Go mux which responds with a 301 status code if the invocation is for `///foo`
+			// This makes Dapr retry with a GET request in all cases, as per specs
+			// See: https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/301
+			{url: fmt.Sprintf("http://localhost:%d/foo", b.daprd1.HTTPPort()), headers: map[string]string{
```

Comment:
Rather than change the test, why can we not just have the previous behavior back?

At _minimum_ we need this test case to be covered, please don't delete it. It's an edge case in which we are interested to know the result of, and we can't have the behavior keep changing on a whim. It also needs to be documented that this is a breaking change if we are not going to maintain the previous behavior of Service Invocation.

Reply:
I guess this could be restored by removing extra slashes at the beginning. Let me see if that can work

Reply:
This has been fixed in #6594. Will update the tests in that PR once this is merged

Reply:
I added a unit test in there. I would rather not make changes to these tests there as once this PR is merged, it will cause more merge conflicts.